packages/stateful/actions/components/ValidatorPicker.tsx

```diff
@@ -0,0 +1,214 @@
+import { ArrowOutward } from '@mui/icons-material'
+import clsx from 'clsx'
+import { useTranslation } from 'react-i18next'
+
+import {
+  Button,
+  CopyToClipboard,
+  FilterableItemPopup,
+  IconButtonLink,
+  InputThemedText,
+  TokenAmountDisplay,
+  Tooltip,
+} from '@dao-dao/stateless'
+import { TokenStake, Validator } from '@dao-dao/types'
+import {
+  convertMicroDenomToDenomWithDecimals,
+  formatPercentOf100,
+  nativeTokenLabel,
+  nativeTokenLogoURI,
+} from '@dao-dao/utils'
+
+export interface ValidatorPickerProps {
+  validators: Validator[]
+  stakes?: TokenStake[]
+  selectedAddress?: string
+  readOnly: boolean
+  onSelect: (validator: Validator) => void
+  // Denom and decimals that correspond with validator.tokens (likely the native
+  // token on the chain).
+  nativeDenom: string
+  nativeDecimals: number
+  displayClassName?: string
+}
+
+export const ValidatorPicker = ({
+  validators,
+  stakes,
+  selectedAddress,
+  readOnly,
+  onSelect,
+  nativeDenom,
+  nativeDecimals,
+  displayClassName,
+}: ValidatorPickerProps) => {
+  const { t } = useTranslation()
+
+  const selectedValidator = validators.find(
+    (validator) => validator.address === selectedAddress
+  )
+
+  // Map validator address to staked amount.
+  const validatorStakedAmountMap = stakes?.reduce(
+    (acc, stake) => ({
+      ...acc,
+      [stake.validator.address]: stake.amount,
+    }),
+    {} as Record<string, number | undefined>
+  )
+
+  // Sort staked first, then by total staked tokens (i.e. voting powe and
+  // popularity).
+  const sortedValidators = [...validators].sort((a, b) => {
+    const aStake = validatorStakedAmountMap?.[a.address]
```

Comment:
What's going on here? This is a bit of a mysterious one-liner to me. I'm sure it gets the job done, but a comment might help a future contributor.